packages/eslint-plugin/scripts/codegen.ts

```diff
@@ -0,0 +1,208 @@
+import type { Rule } from 'eslint'
```

Comment:
This module generates most content inside the eslint plugin, including: rule index, configs, and readme. It is a very simple implementation.